cmd/faUniq/faUniq.go

```diff
@@ -0,0 +1,63 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/fasta"
+	"log"
+)
+
+func faUniq(inFile string, outFile string) {
+	allSeqs := fasta.Read(inFile)
+
+	seqsSeen := make(map[string]int, 0) // This maps the DNA sequence converted to a string (key) to the index in uniqueSeqs (value)
+	seqsSeen[dna.BasesToString(allSeqs[0].Seq)] = 0
+
+	uniqueSeqs := make([]fasta.Fasta, 1)
+	uniqueSeqs[0] = allSeqs[0]
+
+	var currSeq string
+	var seqExists bool
+	var val int
+	for i := 1; i < len(allSeqs); i++ {
+		currSeq = dna.BasesToString(allSeqs[i].Seq)
+		val, seqExists = seqsSeen[currSeq]
+		if !seqExists {
+			seqsSeen[currSeq] = len(uniqueSeqs)
+			uniqueSeqs = append(uniqueSeqs, allSeqs[i])
+		} else {
+			uniqueSeqs[val].Name = uniqueSeqs[val].Name + "; " + allSeqs[i].Name
```

Comment:
I like that in the output fa seq name you include all the names of the same seq in the input fa!